discovery-provider/src/tasks/index.py

```diff
@@ -125,36 +126,58 @@ def initialize_blocks_table_if_necessary(db: SessionManager):
     return target_blockhash
 
 
-def get_latest_block(db: SessionManager):
-    latest_block = None
+def get_latest_block(db: SessionManager, retry: bool = False):
     block_processing_window = int(
         update_task.shared_config["discprov"]["block_processing_window"]
     )
-    with db.scoped_session() as session:
-        current_block_query = session.query(Block).filter_by(is_current=True)
-        assert current_block_query.count() == 1, "Expected SINGLE row marked as current"
+    try:
+        with db.scoped_session() as session:
+            current_block_query = session.query(Block).filter_by(is_current=True)
+            assert (
+                current_block_query.count() == 1
+            ), "Expected SINGLE row marked as current"
 
-        current_block_query_results = current_block_query.all()
-        current_block = current_block_query_results[0]
-        current_block_number = current_block.number
+            current_block_query_results = current_block_query.all()
+            current_block = current_block_query_results[0]
+            current_block_number = current_block.number
 
-        if current_block_number == None:
-            current_block_number = 0
+            if current_block_number == None:
+                current_block_number = 0
 
-        target_latest_block_number = current_block_number + block_processing_window
+            target_latest_block_number = current_block_number + block_processing_window
 
-        latest_block_from_chain = update_task.web3.eth.getBlock("latest", True)
-        latest_block_number_from_chain = latest_block_from_chain.number
+            latest_block_from_chain = update_task.web3.eth.getBlock("latest", True)
+            latest_block_number_from_chain = latest_block_from_chain.number
 
-        target_latest_block_number = min(
-            target_latest_block_number, latest_block_number_from_chain
-        )
+            target_latest_block_number = min(
+                target_latest_block_number, latest_block_number_from_chain
+            )
 
-        logger.info(
-            f"index.py | get_latest_block | current={current_block_number} target={target_latest_block_number}"
-        )
-        latest_block = update_task.web3.eth.getBlock(target_latest_block_number, True)
-    return latest_block
+            logger.info(
+                f"index.py | get_latest_block | current={current_block_number} target={target_latest_block_number}"
+            )
+            latest_block = update_task.web3.eth.getBlock(
+                target_latest_block_number, True
+            )
+
+            # if the block had no transactions, retry after small delay to confirm the block is actually empty
+            # we've seen potential instances of blocks returning no transactions
+            if len(latest_block.transactions) == 0 and not retry:
```

Comment:
semantic ask, can we make this variable retry_count=1 and call get_latest_block(db, retry_count - 1) and then do `if len(latest_block.transactions) == 0 and retry_count > 0:`

i first read this and was confused because it reads "if don't retry, call the fn again"